syslog_rfc5424_parser/message.py

```diff
@@ -75,15 +75,15 @@ def __str__(self):
     def parse(cls, message_string):
         """Construct a syslog message from a string"""
         try:
-            groups = parser.syslog_message.parseString(message_string)
-        except pyparsing.ParseException:
+            groups, _, end_offset = next(parser.syslog_message.scanString(message_string))
+        except (pyparsing.ParseException, StopIteration):
```

Comment:
I'm guessing the `StopIteration` is for pre PEP 479 support?

Reply:
No, it's because `scanString` returns an iterator which is empty if nothing matched at all, and calling `next()` on an empty iterator raises StopIteration.